### ISSUE TITLE
Warn about triple-quoted strings being used as comments

### ISSUE THREAD

**Issue by lordmauve** [PyCQA/pycodestyle#730]
Regarding block comments, PEP8 says [this](https://www.python.org/dev/peps/pep-0008/#block-comments):

> Block comments generally apply to some (or all) code that follows them, and are indented to the same level as that code. 
> Each line of a block comment starts with a # and a single space (unless it is indented text inside the comment).

I sometimes see people writing triple-quoted strings as comments (outside of their proper use, as docstrings):

```python
def some_function():
    """Preen the gobblesnap."""
    gobblesnap = get_gobblesnap()
    """Be careful when preening.
    Gobblesnaps only like it on their left."""
    gobblesnap.preen(left=True, right=False)
```

I believe this stems from novices mistaking the specific case of docstrings for a general comment syntax.

A [straw poll on Twitter rated this as 81% "Yuck"](https://twitter.com/lordmauve/status/968902147923496960). Those who took the time to offer a reply said that this gave the wrong "signals": syntax should convey intention, and this doesn't convey the intention of being a comment, so it adds mental overhead when reading the code.

Would it be possible to issue a warning about this?

**Comment by sigmavirus24** [comment]
> Would it be possible to issue a warning about this?

No. The project is firmly rooted in the document. The document doesn't specifically forbid this and I do recall seeing in past iterations of it that multi-line comments could be written as docstrings.

This is something that a flake8 plugin could easily do

**Comment by sigmavirus24** [comment]
See also https://twitter.com/gvanrossum/status/112670605505077248

For posterity:

> Python tip: You can use multi-line strings as multi-line comments. Unless used as docstrings, they generate no code! :-)

**Comment by sigmavirus24** [comment]
https://stackoverflow.com/a/7696966/1953283 References the above tweet too

**Comment by lordmauve** [comment]
Interesting, thanks.

To me it's still problematic. There are tons of replies on the both the Stack Overflow and tweet pointing out issues with this as a comment syntax (issues I hadn't even thought of).

If Guido's tweet is a BDFL pronouncement about the feature in the language, he doesn't say this within the context of PEP8.

PEP8 doesn't explicitly rule it out, but it does say "Each line of a block comment starts with a #" which is certainly setting an expectation around what block comments are expected to look like.

Maybe PEP8 should be clearer. Perhaps a list post (to python-dev?) would provide clarity?

**Comment by sigmavirus24** [comment]
> Maybe PEP8 should be clearer. Perhaps a list post (to python-dev?) would provide clarity?

You're welcome to bring this up there. I'd ask that you don't drag this thread into it though. People on that list have a history of flaming these discussions, being rude, dismissive, and often display the worst attributes of a community.

If the conversation ever dies down and comes to a conclusion, feel free to report back with what you find.

**Comment by lordmauve** [comment]
Understood. Thanks.
